jiant/__main__.py

```diff
@@ -377,7 +378,8 @@ def initial_setup(args, cl_args):
     random.seed(seed)
     torch.manual_seed(seed)
     log.info("Using random seed %d", seed)
-    if args.cuda >= 0:
+    if isinstance(args.cuda, int) and args.cuda >= 0:
```

Comment:
shouldn't this cuda check also happen if you're trying to auto or use multiple GPUs ?

Reply:
parse_cuda_args_list converts auto -> range(len(devices)), basically using all devices on the machine.